### ISSUE TITLE
Calling `copy` on an instance of `URL` generates a warning

### ISSUE THREAD

**Issue by rec** [sqlalchemy/sqlalchemy#7400]
### Describe the bug

Calling the Python built-in `copy.copy` on an instance of `sqlalchemy.engine.url.URL` triggers the warning on or around `sqlalchemy/engine/url.py`

### To Reproduce

```python
copy.copy(URL.create('postgresql+psycopg2'))
```

### Error

```
/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/copyreg.py:91: SADeprecationWarning: Calling URL() directly is deprecated and will be disabled in a future release.  The public constructor for URL is now the URL.create() method.
  return cls.__new__(cls, *args)
``` 

### Versions

- OS: Darwin Kernel Version 19.6.0
- Python: Python 3.8.10
- SQLAlchemy: "1.4.27
- Database: Postgresql 12.x
- DBAPI (eg: psycopg, cx_oracle, mysqlclient): psycopg

### Additional context

Adding this method to `sqlalchemy.engine.url.URL` made the problem go away:

```
    def __copy__(self):
        return __class__.create(
            self.drivername,
            self.username,
            self.password,
            self.host,
            self.port,
            self.database,
            self.query,
        )
```

Thanks for all your hard-work on our behalf, and also for your manicured clean code.

**Comment by zzzeek** [comment]
we can accept PRs for this which includes a test in test/engine/test_parseconnect.py

**Comment by rec** [comment]
On it.

**Comment by sqla-tester** [comment]
**Tom Ritchford** has proposed a fix for this issue in the **main** branch:

**Add __copy__, __deepcopy__ to URL. Fixes: #7400** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3352

**Comment by sqla-tester** [comment]
**Tom Ritchford** has proposed a fix for this issue in the **rel_1_4** branch:

**Add __copy__, __deepcopy__ to URL. Fixes: #7400** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3353
